src/main/java/uk/gov/hmcts/sptribs/recordlisting/LocationService.java

```diff
@@ -51,46 +50,50 @@ public DynamicList getAllRegions() {
     }
 
     private Region[] getRegions() {
-        ResponseEntity<Region[]> regionResponseEntity = null;
 
         try {
-            regionResponseEntity = locationClient.getRegions(
+            List<Region> list = locationClient.getRegions(
                 authTokenGenerator.generate(),
                 httpServletRequest.getHeader(AUTHORIZATION),
                 REGION_ALL);
+            if (CollectionUtils.isEmpty(list)) {
+                return null;
+            }
+            Region[] regions = new Region[list.size()];
+            for (int i = 0; i < list.size(); i++) {
+                regions[i] = list.get(i);
+            }
+            return regions;
         } catch (FeignException exception) {
             log.error("Unable to get Region data from reference data with exception {}",
                 exception.getMessage());
         }
 
-        return Optional.ofNullable(regionResponseEntity)
-            .map(response ->
-                Optional.ofNullable(response.getBody())
-                    .orElseGet(() -> null)
-            )
-            .orElseGet(() -> null);
+        return null;
     }
 
     private HearingVenue[] getCourtVenues(String regionId) {
-        ResponseEntity<HearingVenue[]> hearingVenueResponseEntity = null;
 
         try {
-            hearingVenueResponseEntity = locationClient.getHearingVenues(
+            List<HearingVenue> list = locationClient.getHearingVenues(
                 authTokenGenerator.generate(),
                 httpServletRequest.getHeader(AUTHORIZATION),
                 regionId,
                 "Y");
+            if (CollectionUtils.isEmpty(list)) {
+                return null;
+            }
+            HearingVenue[] hearingVenues = new HearingVenue[list.size()];
+            for (int i = 0; i < list.size(); i++) {
+                hearingVenues[i] = list.get(i);
+            }
+            return hearingVenues;
```

Comment:
this block can be shortened to `return list.toArray(new HearingVenue[list.size()]);`

Reply:
İ started that way, but got error on returnig